zebra-chain/src/orchard/shielded_data.rs

```diff
@@ -48,6 +50,37 @@ impl ShieldedData {
     pub fn nullifiers(&self) -> impl Iterator<Item = &Nullifier> {
         self.actions().map(|action| &action.nullifier)
     }
+
+    /// Calculate the Action binding verification key.
+    ///
+    /// Getting the binding signature validating key from the Spend and Output
+    /// description value commitments and the balancing value implicitly checks
+    /// that the balancing value is consistent with the value transfered in the
+    /// Spend and Output descriptions but also proves that the signer knew the
+    /// randomness used for the Spend and Output value commitments, which
+    /// prevents replays of Output descriptions.
+    ///
+    /// The net value of Spend transfers minus Output transfers in a transaction
+    /// is called the balancing value, measured in zatoshi as a signed integer
+    /// v_balance.
+    ///
+    /// Consistency of v_balance with the value commitments in Spend
+    /// descriptions and Output descriptions is enforced by the binding
+    /// signature.
+    ///
+    /// Instead of generating a key pair at random, we generate it as a function
+    /// of the value commitments in the Spend descriptions and Output
+    /// descriptions of the transaction, and the balancing value.
+    ///
+    /// https://zips.z.cash/protocol/protocol.pdf#orchardbalance
```

Comment:
The description is pretty much the same as sapling because i think it will work but we should probably not write the exact same thing.

Reply:
This description needs to be updated based on the Orchard consensus rules:
https://zips.z.cash/protocol/protocol.pdf#actiondesc
https://zips.z.cash/protocol/protocol.pdf#orchardbalance
https://zips.z.cash/protocol/protocol.pdf#actionencodingandconsensus

Specifically:
* In some contexts, Orchard `Action`s replace Sapling `Spend`s and `Output`s
* In other contexts, Orchard input notes replace Sapling `Spend`s, and Orchard ouput notes replace Sapling `Output`s
* The key generation is different and needs to be based on the Orchard consensus rules

Reply:
This seems to be a relevant part of the spec:

> Consistency of vbalanceOrchard with the value commitments in Action descriptions is enforced by the Orchard binding signature. The rôle of this signature in the Orchard protocol is to prove that the net value spent (i.e. the total value spent minus the total value produced) by Action transfers is consistent with the vbalanceOrchard field of the transaction.
> Non-normative note: The other rôle of Sapling binding signatures, to prove that the signer knew the randomness used for commitments in order to prevent them from being replayed, is less important in Orchard because all Action descriptions have a spend authorization signature. Still, an Orchard binding signature does prove that the signer knew this commitment randomness; this provides defence in depth and reduces the differences of Orchard from Sapling, which may simplify security analysis.
> Instead of generating a key pair at random, we generate it as a function of the value commitments in the Action descriptions of the transaction, and the Orchard balancing value.

https://zips.z.cash/protocol/protocol.pdf#orchardbalance

Reply:
@teor2345 I suggested several updates to these docs above 🎈